packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:ui';
+
+import '../../../components.dart';
+import '../../palette.dart';
+
+/// Adds a collection of paints to a component
+///
+/// Component will always have a main Paint that can be accessed
+/// by the [paint] attribute and other paints can be manipulated/accessed
+/// using [getPaint], [setPaint] and [deletePaint]
+mixin HasPaint<T extends Object> on BaseComponent {
+  final Map<T, Paint> _paints = {};
+
+  Paint paint = BasicPalette.white.paint();
+
+  /// Gets a paint from the collection, returns the main paint if no [key] is provided
+  Paint getPaint([T? key]) {
+    if (key == null) {
+      return paint;
+    }
+
+    final _paint = _paints[key];
+
+    if (_paint == null) {
+      throw ArgumentError('No Paint found for $key');
+    }
+
+    return _paint;
+  }
+
+  /// Sets a paint on the collection
+  void setPaint(T key, Paint paint) => _paints[key] = paint;
```

Comment:
why no make key nullable here too (and set `paint`)?

Reply:
I could, but I don't see much point since `paint` is an attribute now. Also to make it nullable, we would need to make it named, so we would need to write more.

Unsure, no hard opinions here, I think we don't need to do that, but if you really think it will be good, I can implement.

Reply:
yeah I think you are right, not much point in having